cityofaustin/atd-moped
# Edit drawn component features

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10958
https://github.com/cityofaustin/atd-data-tech/issues/10992

This PR updates the component map UI so we can edit drawn line and point features. This adds a new component called `UpdateComponentDrawTools` that is in charge of adding, updating, and deleting features when editing an existing component. This component dispatches new reducer actions that are added in `useUpdateComponents`. It also updates the styles and colors of component features when creating and editing features.

## Testing
**URL to test:** 

local or https://md-10958-edit-drawn-features--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Create a new project
2. Create one component of line type and then select and draw some line segments. Notice that the new segments (drawn or selected) are highlighted with orange colors.
3. After saving, select the component from the list and click "Edit" and then "Map"
4. You'll see the existing features highlighted in a bright orange.
5. Deselect an existing line segment to remove it from the component and then save the change.
6. The edited component will update and you should no longer see the segment that you removed in the features of the component that you edited.
7. Now, edit the same component, but this time click a drawn line to activate the edit draw tools and see the vertices circled in green.
8. Click and hold on the selected line and drag it to a new location. Then, save.
9. You'll see the updated location saved in the map.
10. Now, edit the same component features and delete the drawn feature by selecting it and then clicking the trash can icon. Then, save.
11. You'll see the line is now removed from the features of the component that you edited.
12. Repeat the steps above with a component of point type.

---
#### Ship list
